core/generator.js

```diff
@@ -113,13 +118,25 @@ Blockly.Generator.prototype.workspaceToCode = function(workspace) {
       code.push(line);
     }
   }
-  code = code.join('\n');  // Blank line between each section.
-  code = this.finish(code);
-  // Final scrubbing of whitespace.
-  code = code.replace(/^\s+\n/, '');
-  code = code.replace(/\n\s+$/, '\n');
-  code = code.replace(/[ \t]+\n/g, '\n');
-  return code;
+  if(blocksNotGenerated.length == 0){
+    code = code.join('\n');  // Blank line between each section.
+    code = this.finish(code);
+    // Final scrubbing of whitespace.
+    code = code.replace(/^\s+\n/, '');
+    code = code.replace(/\n\s+$/, '\n');
+    code = code.replace(/[ \t]+\n/g, '\n');
+    return code;
+  }
+  else{
+    var textToBeLogged = 'The generator code for the following blocks not specified for ' + this.name_ + ': \n';
+    
+    //Blocks not specified traversed
+    for(var x = 0; x < blocksNotGenerated.length; x++){
+      textToBeLogged +=  '* ' + blocksNotGenerated[x].type + '\n';
+    }
+    textToBeLogged += 'Fix those to be able to generate code';
+    throw textToBeLogged;
```

Comment:
I have to discuss with others what the implication of this throw is. Specifically, what are the implications to an existing app that has not been erroring in the past and upgrades to this version of Blockly that now starts emitting errors.
@rachel-fenichel  ?